### PR TITLE
Change HTTP projects to target netcoreapp3.0

### PR DESCRIPTION
**Pull request by Tratcher**
#3754
- Microsoft.AspNetCore.Http.Features and tests have been excluded as they're used by the signalr client. See https://github.com/aspnet/AspNetCore/issues/3754#issuecomment-443049198
- Pipes failures: https://github.com/aspnet/AspNetCore/issues/4621

### CODE REVIEW

**Comment by natemcmaster** [comment]
I think you'll need to update Razor first.

**Comment by Tratcher** [comment]
Reverted the HTML changes so those can be done with Razor.

File: src/Http/Headers/src/CacheControlHeaderValue.cs [hunk 1]
```diff
@@ -652,7 +652,7 @@ private static void AppendValues(StringBuilder sb, IEnumerable<StringSegment> va
                     sb.Append(", ");
                 }
 
-                sb.Append(value);
+                sb.Append(value.AsSpan());
```
Comment (JamesNK) [inline src/Http/Headers/src/CacheControlHeaderValue.cs]: For my edification, what is the benefit of StringBuilder.Append with a Span over a String?

Comment (Tratcher) [reply src/Http/Headers/src/CacheControlHeaderValue.cs]: value isn't a string, it's a StringSegment that has implicit converters to Span and Memory. In 3.0 Append added Span and Memory overloads and now we get ambiguity errors.

**Comment by Tratcher** [comment]
Ready for review. The test failures are known issues in other components.

File: src/Http/Http/src/StreamPipeReader.cs [hunk 2]
```diff
@@ -227,7 +227,7 @@ public override async ValueTask<ReadResult> ReadAsync(CancellationToken cancella
                 try
                 {
                     AllocateReadTail();
-#if NETCOREAPP2_2
+#if NETCOREAPP3_0
                     var length = await _readingStream.ReadAsync(_readTail.AvailableMemory.Slice(_readTail.End), tokenSource.Token);
 #elif NETSTANDARD2_0
```
Comment (natemcmaster) [inline src/Http/Http/src/StreamPipeReader.cs]: `#if NETSTANDARD2_0` can be removed now, right?

Comment (Tratcher) [reply src/Http/Http/src/StreamPipeReader.cs]: We're keeping it because this code is going to get moved to corefx and support netstandard.

**Review by JunTaoLuo — APPROVED** [review]

**Pull request merged**